modules/mobileApp/routes/agendas.ts

```diff
@@ -13,18 +15,46 @@ const router = express.Router();
 router.get('/agendasDisponibles', async (req: any, res, next) => {
     const pipelineAgendas = [];
     const matchAgendas = {};
+    const condiciones: any = await CondicionPaciente.find({activo: true});
+    let reglas = [];
+    let fieldRegla;
+    if (req.query.idPaciente) {
+        for (const condicion of condiciones) {
+            const verificar = await verificarCondicionPaciente(condicion, req.query.idPaciente);
+            if (verificar) {
+                reglas.push(condicion.tipoPrestacion.conceptId);
+            }
+        }
+    }
+
     if (req.query.conceptId) {
         matchAgendas['tipoPrestaciones.conceptId'] = req.query.conceptId;
     }
-
     matchAgendas['horaInicio'] = { $gt: new Date(moment().format('YYYY-MM-DD HH:mm')) };
     matchAgendas['bloques.restantesProgramados'] = { $gt: 0 };
-    matchAgendas['bloques.restantesMobile'] = { $gt: 0 };
-
     matchAgendas['estado'] = 'publicada';
     matchAgendas['dinamica'] = false;
-
+    if (reglas) {
+        matchAgendas['$or'] = [
+            {
+                'bloques.restantesMobile': { $gt: 0 },
+            },
+            {
+                'tipoPrestaciones.conceptId': { $in:  reglas  }
+            }
+        ];
+        fieldRegla =  {
+            cumpleRegla: {
+                $not: [{$eq: [{$size: { $setIntersection: ['$tipoPrestaciones.conceptId', reglas]}}, 0]}]
```

Comment:
Capaz que un **$in** es más vistoso: https://docs.mongodb.com/manual/reference/operator/aggregation/in/

Reply:
Como lo tenías?